### PR TITLE
fix(sentry): log error boundary errors to Sentry

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Logging `ErrorBoundary` errors to Sentry

- Updated sentry to latest
- ~Enabled source maps to improve Sentry error readability~ Disabled as it was breaking the Vercel build :(
- Removed ErrorBoudaryMod file

Sample Sentry error: https://cowprotocol.sentry.io/issues/4415065870/events/a3f15eee22fc43f3a3fbb230ad99e2d7/
image

# To Test

1. Open limit orders
2. Pick a BUY token
* App WILL CRASH!!!
* Error should be logged in Sentry

# Notes

- ⚠️ DO NOT MERGE BEFORE REVERTING THE COMMIT 39bad8ca3643a35e137d00988dfe5a4a74e52553 ⚠️
- `SENTRY_AUTH_TOKEN` GithubActions envvar might need to be updated

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/limitOrders/containers/LimitOrdersWidget/index.tsx [hunk 1]
```diff
@@ -163,6 +164,10 @@ const LimitOrders = React.memo((props: LimitOrdersProps) => {
   const inputCurrency = inputCurrencyInfo.currency
   const outputCurrency = outputCurrencyInfo.currency
 
+  useEffect(() => {
+    if (outputCurrency) BigNumber.from('break!!!')
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/limitOrders/containers/LimitOrdersWidget/index.tsx:168]: Don't forget to remove this

File: apps/cowswap-frontend/src/legacy/components/ErrorBoundary/index.tsx
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/components/ErrorBoundary/index.tsx:1]: Sorry for the noise, removed the mod file

File: apps/cowswap-frontend/src/legacy/components/ErrorBoundary/index.tsx [hunk 2]
```diff
@@ -1,2 +1,152 @@
...
+              <Wrapper>{isChunkLoadError ? <ChunkLoadError /> : <ErrorWithStackTrace error={error} />}</Wrapper>
+              <FooterWrapper>
+                <Footer />
+              </FooterWrapper>
+            </AppWrapper>
+          )
+        }}
+      >
+        {this.props.children}
+      </Sentry.ErrorBoundary>
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/components/ErrorBoundary/index.tsx:149]: This is the main change.
Wrapped the whole component response in Sentry.ErrorBoundary and put the fallback inside of it

File: apps/cowswap-frontend/vite.config.ts [hunk 3]
```diff
@@ -119,6 +121,12 @@ export default defineConfig(({ mode }) => {
           globPatterns: ['**/*.{js,css,html,png,jpg,svg,json,woff,woff2,md}'],
         },
       }),
+      sentryVitePlugin({
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply apps/cowswap-frontend/vite.config.ts:124]: Hmm, I though it has been working before with Webpack

[the remark this answers is not in the document]
Comment (alfetopito) [reply apps/cowswap-frontend/vite.config.ts:124]: I'm not sure.
Sentry was working before this change, but the difference is that we were only explicitly sending errors to sentry without the catch all in the ErrorBoundary.

And this part is not related to that either, as this plugin was added mostly for the sourcemap generation:

https://www.npmjs.com/package/@sentry/vite-plugin 

> A Vite plugin that provides source map and release management support for Sentry.

File: apps/cowswap-frontend/vite.config.ts [hunk 4]
```diff
@@ -74,6 +75,7 @@ export default defineConfig(({ mode }) => {
     },
 
     build: {
+      sourcemap: true,
```
[the remark this answers is not in the document]
Comment (alfetopito) [reply apps/cowswap-frontend/vite.config.ts]: Disabling this resulted in vercel builds to succeed. We will not have source maps on Sentry for now :(

**Review by shoom3301 — APPROVED** [review]
@alfetopito great job!
Please, don't forget about this: https://github.com/cowprotocol/cowswap/pull/3077/files#r1303055477

**Pull request merged**